### ISSUE TITLE
Top Navigation:  Strange behaviour with Grouped Navigation

### ISSUE THREAD

**Issue by GCHQ-Developer-299** [mi6/ic-ui-kit#1873]
## Summary of the bug 

In recent screenreader testing we found a couple of strange behaviours with [https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-top-navigation--with-grouped-navigation](Top Nav Grouped Navigation)
- Grouped Navigation buttons have aria-expanded=true even on load
- Grouped Navigation buttons cannot be 'clicked' by Voiceover

Expected outcomes:
- [ ] aria-expanded should be set to false on first load, and should update (it currently seems to remain 'true' regardless of user interaction) 
- [x] The navigation group buttons should be 'clickable' for Voiceover (see below)

## 🪜 How to reproduce 
In Voiceover you can replicate clicks with Ctrl+Option+Space, and this 'click' doesn't open the Expanded Navigation groups. They still open when Enter is pressed, but this is the only button in our library that has this behaviour. 

## 📸 Screenshots or code 
screenshot of the rendered group navigation - the tab index and aria-expanded seem curious and might be the reason for the two issues.
![Screenshot 2024-05-16 at 15 44 20](https://github.com/mi6/ic-ui-kit/assets/111292477/857c3552-e266-450d-b99c-3b05f6c31cc5)

**Comment by GCHQ-Developer-847** [comment]
I have removed the tabindex as part of #3259 (PR #3378) and it looks like this has fixed the Ctrl+Option+Space issue! See the updated ["With grouped navigation" story](https://mi6.github.io/ic-ui-kit/branches/fix/3259-mobile-top-nav-group-title-looks-clickable/web-components/?path=/story/web-components-top-navigation--with-grouped-navigation).

However, the second issue about `aria-expanded` still needs to be resolved. Also, I have noticed that `aria-expanded` does occasionally get set correctly: if you open the navigation group using Space or Enter, tab off it and then tab back onto it, it will say it is collapsed, but do the same again and it will say it is expanded when it isn't.

But yes it never seems to be set correctly on load, or if you open the group by click (or Ctrl+Option+Space, even though it's now expanding the dropdown menu).

**Comment by GCHQ-Developer-847** [comment]
Marked the second bullet point as done to reflect my comment above. Also just had another look at the expanded screen reader behaviour and it never seems to say it's collapsed at all now - just always expanded

**Comment by GCHQ-Developer-847** [comment]
In a draft PR as it builds upon changes from PR #3378
